### ISSUE TITLE
bundling failed: Error: Unable to resolve module `AccessibilityInfo`

### ISSUE THREAD

**Issue by ardimh7** [GeekyAnts/NativeBase-KitchenSink#124]
I've following the cloning guide on the readme.md
the build ran successfully but I encountered this error:

<pre>
Loading dependency graph, done.
error: bundling failed: Error: Unable to resolve module `AccessibilityInfo` from `E:\Smartsoft\tugasReact\NativeBase-KitchenSink\node_modules\react-native\Libraries\react-native\react-native-implementation.js`: Module `AccessibilityInfo` does not exist in the Haste module map

This might be related to https://github.com/facebook/react-native/issues/4968
To resolve try the following:
  1. Clear watchman watches: `watchman watch-del-all`.
  2. Delete the `node_modules` folder: `rm -rf node_modules && npm install`.
  3. Reset Metro Bundler cache: `rm -rf /tmp/metro-bundler-cache-*` or `npm start -- --reset-cache`.  4. Remove haste cache: `rm -rf /tmp/haste-map-react-native-packager-*`.
    at ModuleResolver.resolveDependency (E:\Smartsoft\tugasReact\NativeBase-KitchenSink\node_modules\metro\src\node-haste\DependencyGraph\ModuleResolution.js:167:1306)
    at ResolutionRequest.resolveDependency (E:\Smartsoft\tugasReact\NativeBase-KitchenSink\node_modules\metro\src\node-haste\DependencyGraph\ResolutionRequest.js:80:16)
    at DependencyGraph.resolveDependency (E:\Smartsoft\tugasReact\NativeBase-KitchenSink\node_modules\metro\src\node-haste\DependencyGraph.js:237:485)
    at Object.resolve (E:\Smartsoft\tugasReact\NativeBase-KitchenSink\node_modules\metro\src\lib\transformHelpers.js:116:25)
    at dependencies.map.result (E:\Smartsoft\tugasReact\NativeBase-KitchenSink\node_modules\metro\src\DeltaBundler\traverseDependencies.js:298:29)
    at Array.map (<anonymous>)
    at resolveDependencies (E:\Smartsoft\tugasReact\NativeBase-KitchenSink\node_modules\metro\src\DeltaBundler\traverseDependencies.js:294:16)
    at E:\Smartsoft\tugasReact\NativeBase-KitchenSink\node_modules\metro\src\DeltaBundler\traverseDependencies.js:159:33
    at Generator.next (<anonymous>)
    at step (E:\Smartsoft\tugasReact\NativeBase-KitchenSink\node_modules\metro\src\DeltaBundler\traverseDependencies.js:239:307)
 BUNDLE  [android, dev] ..\..\../index.js ░░░░░░░░░░░░░░░░ 0.0% (0/2), failed.
</pre>

is there any solution of my problem? thanks before

**Comment by ardimh7** [comment]
![whatsapp image 2018-07-12 at 5 53 25 pm](https://user-images.githubusercontent.com/16507813/42629160-aba3e91e-85fc-11e8-9bcb-c8d083fd6d2a.jpeg)

**Comment by SupriyaKalghatgi** [comment]
Thats react native issue, follow the steps mentioned in this red screen

**Comment by SupriyaKalghatgi** [comment]
@ardimh7 Did you try?

**Comment by eoamusan** [comment]
Why is this being closed? What solution has been proferred?

**Comment by Parisasuchdev** [comment]
This is the version mismatch issue :( you have to downgrade react-native version to 0.55.4. in package.json file then delete node modules folder and then run this command
`npm install --save`

this works for many, but not for me.
it's giving me vulnerability.

**Comment by chenSunRise** [comment]
Try to check the port

**Comment by TarunShashank** [comment]
@Parisasuchdev 
I am also getting some error on following the workaround. 

What I did:

`C:\Users\Lenovo\Desktop\React projects\MyProject>npm install --save
npm WARN react-native@0.55.4 requires a peer of react@16.3.1 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! Maximum call stack size exceeded

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Lenovo\AppData\Roaming\npm-cache\_logs\2018-08-03T05_19_39_386Z-debug.log

C:\Users\Lenovo\Desktop\React projects\MyProject>`

Any solution for this

**Comment by devchirag07** [comment]
Its some incompatible versioning problems in React native.Try this it will work
1. Change versions of following in package.json 
    In dependencies :-
             "react": "16.3.1"
             "react-native": "0.55.4"
   In devDependencies :-
               "babel-preset-react-native": "2.1.0"
2. Delete node_modules and run npm install --save
3. Then run application by react-native run-android

It will run fine ...

**Comment by Mohsens22** [comment]
I have the same issue. Someone reopen this

**Comment by pradeepchaudhari91** [comment]
I have the same issue.

**Comment by nucklearproject** [comment]
This patch resolve the issue in RN 0.56 https://github.com/facebook/react-native/issues/20460#issuecomment-411858190

Apply the patch and reset the cache. 
rm -rf $TMPDIR/metro-* && rm -rf $TMPDIR/react-* && rm -rf $TMPDIR/haste-* 

if dont work in your android folder
gradlew clean

**Comment by Usamaliaquat123** [comment]
It seems like a bug in 0.56 related to dependencies. The "solution" is to find the correct combination of dependencies' versions. We found a workaround by installing those versions EXACTLY:

```
react-native >> 0.55.4
babel-core >> latest 
babel-loader >> latest
babel-preset-react-native >> 4.0.0
```
So you have to run those commands in order:

react-native init AwesomeProject
```
cd AwesomeProject
react-native run-android
npm uninstall react-native
npm install --save react-native@0.55.4
react-native run-android
npm install --save babel-core@latest babel-loader@latest
npm uninstall --save babel-preset-react-native
npm install --save babel-preset-react-native@4.0.0
react-native run-android
```
Its worked 👍

**Comment by Shenith** [comment]
thnx devchirag07

**Comment by zoiec** [comment]
I had this error and just needed to update xcode and command line tools

**Comment by SirPhemmiey** [comment]
There's a bug in the recent version of react-native. To fix (at least temporarily) the problem use the following version of react and react native.

```
"react": "^16.4.1",
 "react-native": "^0.55.4"
```

**Comment by Vincent1712** [comment]
I think it haven't necessary to delete 'node_module' file cuz it's just close all application or restart computer to full delete:

**Step 1**: 
**In package.json:**
                         **In dependencies :-**
                         "react": "16.4.1"
                         "react-native": "0.55.4"
                          **In devDependencies :-**
                         "babel-preset-react-native": "4.0.0"

**Step 2**
 npm install --save 
 react-native run-android 
 and it's just have met error **unable to load script from assets 'index.android.bundle'**  ==> go to **Step 3**

**Step 3**: 
1.	mkdir android\app\src\main\assets
2.	react-native bundle --platform android --dev false --entry-file index.js --bundle-output android/app/src/main/assets/index.android.bundle --assets-dest android/app/src/main/res
3.	react-native run-android

Hope to useful. Cheer!

**Comment by aujisti-ador** [comment]
@devchirag07 thanks bro,it works. [](https://github.com/GeekyAnts/NativeBase-KitchenSink/issues/124#issuecomment-411655989l)

**Comment by cemelma** [comment]
@Vincent1712 Thanks your solve.

**Comment by c0mm4nDer** [comment]
@Vincent1712 & @devchirag07 Thanks to both of you

**Comment by baicuhaha** [comment]
@Vincent1712  thank you！

**Comment by asiriPiyajanaka** [comment]
@devchirag07 ThankYou..You saved my day.

**Comment by asiriPiyajanaka** [comment]
@devchirag07  worked for me..

**Comment by donaldCull** [comment]
Does anyone have a solution that doesn't involve using older versions of react, rn and other dependencies?

**Comment by bipulthapa** [comment]
@devchirag07  Thanks dude it worked !!

**Comment by tvanhuu** [comment]
> Its some incompatible versioning problems in React native.Try this it will work
> 
> 1. Change versions of following in **package.json**
>    In dependencies :-
>    **"react": "16.3.1"
>    "react-native": "0.55.4"**
>    In devDependencies :-
>    **"babel-preset-react-native": "2.1.0"**
> 2. Delete **node_modules** and run **npm install --save**
> 3. Then run application by **react-native run-android**
> 
> It will run fine ...

Thank bro. Worked for me ^^. Nice day

**Comment by SubashManian** [comment]
> It seems like a bug in 0.56 related to dependencies. The "solution" is to find the correct combination of dependencies' versions. We found a workaround by installing those versions EXACTLY:
> 
> ```
> react-native >> 0.55.4
> babel-core >> latest 
> babel-loader >> latest
> babel-preset-react-native >> 4.0.0
> ```
> So you have to run those commands in order:
> 
> react-native init AwesomeProject
> 
> ```
> cd AwesomeProject
> react-native run-android
> npm uninstall react-native
> npm install --save react-native@0.55.4
> react-native run-android
> npm install --save babel-core@latest babel-loader@latest
> npm uninstall --save babel-preset-react-native
> npm install --save babel-preset-react-native@4.0.0
> react-native run-android
> ```
> Its worked 👍

I stuck in the issue of module Accessibility and searched solution for this issue in GitHub lot of members says downgrade the react-native version i try that too still i got the issue. when i try to upgrade the version i got the issues in my project files are missed. it says the Unable to resolve ../screens/hub.js(my project file)

I thought you already faced this issue it you give me some solution to fix this.

This are my dependencies.
<
"devDependencies": {
    "@babel/plugin-proposal-optional-catch-binding": "^7.0.0",
    "babel-jest": "23.4.2",
    "babel-preset-react-native": "^5",
    "jest": "23.5.0",
    "react-test-renderer": "16.4.1"
  },
"dependencies": {
    "@expo/vector-icons": "^6.3.1",
    "axios": "^0.18.0",
    "immutable": "^3.8.2",
    "lodash": "^4.17.11",
    "moment": "^2.22.2",
    "native-base": "^2.8.0",
    "react": "16.4.1",
    "react-native": "0.56.0",
    "react-native-actionsheet": "^2.4.2",
    "react-native-auth0": "^1.3.0",
    "react-native-hub-bridge": "file:react-native-hub-bridge",
    "react-native-i18n": "^2.0.15",
    "react-native-loading-spinner-overlay": "^0.5.2",
    "react-native-modal": "^6.5.0",
    "react-native-modal-dropdown": "^0.6.2",
    "react-native-network-info": "^4.0.0",
    "react-native-progress": "^3.5.0",
    "react-native-select-multiple": "^1.2.0",
    "react-native-sortable-grid": "^2.0.0",
    "react-native-super-grid": "^2.4.3",
    "react-navigation": "2.5.3",
    "react-navigation-tabs": "^0.8.2",
    "react-redux": "^5.0.7",
    "redux": "^4.0.0",
    "redux-act": "^1.7.4",
    "redux-form": "^7.4.2",
    "redux-logger": "^3.0.6",
    "redux-persist": "^5.10.0",
    "redux-saga": "^0.16.0",
    "reselect": "^3.0.1",
    "xcode": "^1.0.0"
  }
>

**Comment by moohat** [comment]
> Its some incompatible versioning problems in React native.Try this it will work
> 
> 1. Change versions of following in **package.json**
>    In dependencies :-
>    **"react": "16.3.1"
>    "react-native": "0.55.4"**
>    In devDependencies :-
>    **"babel-preset-react-native": "2.1.0"**
> 2. Delete **node_modules** and run **npm install --save**
> 3. Then run application by **react-native run-android**
> 
> It will run fine ...

thank you for your sharing, it works for me, i only install babel-preset-react-native and delete node_modules and run npm install --save.
